removing extra tests, as they all raise the same error of min number of classes

VariableVasasMT opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
none
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.

removing extra tests, that raised the same errors

#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/scikit-learn/scikit-learn/15069)
<!-- Reviewable:end -->

VariableVasasMT committed:
removing extra tests, as they all raise the same error of min number of classes

VariableVasasMT commented:
@raghavrv please check, i was working on https://github.com/machinelearnjs/machinelearnjs/pull/243, found that these tests check the same things as the one above it, don't think it should exist.

VariableVasasMT committed:
moving tests in correct order, as they all raise the same error of min number of classes

VariableVasasMT committed:
moving tests in correct order, as they all raise the same error of min number of classes

amueller commented:
Sorry, can you please clarify what you're fixing?
@raghavrv is not longer contributing.

VariableVasasMT commented:
@amueller I forgot the exact change but 
`a = StratifiedShuffleSplit(3, 0.2).split(X, y)`

`next(a)`

**we get error:**
>>>Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/mnt/code/anaconda3/lib/python3.7/site-packages/sklearn/model_selection/_split.py", line 1315, in split
    for train, test in self._iter_indices(X, y, groups):
  File "/mnt/code/anaconda3/lib/python3.7/site-packages/sklearn/model_selection/_split.py", line 1695, in _iter_indices
    raise ValueError("The least populated class in y has only 1"
ValueError: The least populated class in y has only 1 member, which is too few. The minimum number of groups for any class cannot be less than 2.
`


`b = StratifiedShuffleSplit(3, 2).split(X, y)`

`next(b)`

**we get the same error**

>>>Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/mnt/code/anaconda3/lib/python3.7/site-packages/sklearn/model_selection/_split.py", line 1315, in split
    for train, test in self._iter_indices(X, y, groups):
  File "/mnt/code/anaconda3/lib/python3.7/site-packages/sklearn/model_selection/_split.py", line 1695, in _iter_indices
    raise ValueError("The least populated class in y has only 1"
ValueError: The least populated class in y has only 1 member, which is too few. The minimum number of groups for any class cannot be less than 2.

**but if:**
`X = np.arange(9)`
`y = np.asarray([0, 0, 0, 1, 1, 1, 2, 2, 2])`
`c = StratifiedShuffleSplit(3, 2).split(X, y)`
`next(c)`

**we get error** 
>>>Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/mnt/code/anaconda3/lib/python3.7/site-packages/sklearn/model_selection/_split.py", line 1315, in split
    for train, test in self._iter_indices(X, y, groups):
  File "/mnt/code/anaconda3/lib/python3.7/site-packages/sklearn/model_selection/_split.py", line 1707, in _iter_indices
    (n_test, n_classes))
ValueError: The test_size = 2 should be greater or equal to the number of classes = 3

which is the same error as said in comment `# Check that error is raised if the test set size is smaller than n_classes`

so I think that should be the order of test cases.
:smile:

VariableVasasMT commented:
@amueller please review.

jnothman commented:
I'm not really understanding what this is about either. Why do we need to
change the tests?

VariableVasasMT commented on sklearn/model_selection/tests/test_split.py:
```diff
@@ -633,16 +633,17 @@ def test_stratified_shuffle_split_init():
+    X = np.arange(9)
+    y = np.asarray([0, 0, 0, 1, 1, 1, 2, 2, 2])
+
     # Check that error is raised if the test set size is smaller than n_classes
```
@jnothman the test written below wasn't testing what comment said.

VariableVasasMT replied:
of course I might be wrong in my assumption, I am not sure anymore!

thomasjpfan replied:
To avoid confusion in the future, we can use

```py
msg = "The test_size = 2 should be greater or equal to the number of classes = 3"
with pytest.raises(ValueError, match=msg):
    next(StratifiedShuffleSplit(3, 2).split(X, y))
```

so we can check for the error message explicitly.

VariableVasasMT replied:
will add these to the messages!

cmarmo commented:
There is no longer `assert_raises` in [test_split.py](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/model_selection/tests/test_split.py), which has been largely reorganized in the meanwhile.
I am closing this one.
